docs/LoaderImplementationInterface.md

```diff
@@ -0,0 +1,1081 @@
+
+```
+HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Class\{Adapter GUID}\000X\VulkanDriverNameWow
+HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Class\{SoftwareComponent GUID}\000X\VulkanDriverNameWow
+```
+
+If any of the above values exist and is of type `REG_SZ`, the loader will open
+the JSON manifest file specified by the key value.
+Each value must be a full absolute path to a JSON manifest file.
+The values may also be of type `REG_MULTI_SZ`, in which case the value will be
+interpreted as a list of paths to JSON manifest files.
+
+Additionally, the Vulkan loader will scan the values in the following Windows
+registry key:
+
+```
+HKEY_LOCAL_MACHINE\SOFTWARE\Khronos\Vulkan\Drivers
+```
+
+For 32-bit applications on 64-bit Windows, the loader scan's the 32-bit
+registry location:
+
+```
+HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Khronos\Vulkan\Drivers
+```
+
+Every implementation in these locations should be given as a DWORD, with value
+0, where the name of the value is the full path to a JSON manifest file.
+The Vulkan loader will attempt to open each manifest file to obtain the
+information about an implementation's shared library (".dll") file.
+
+For example, let us assume the registry contains the following data:
+
+```
+[HKEY_LOCAL_MACHINE\SOFTWARE\Khronos\Vulkan\Drivers\]
+
+"C:\vendor a\vk_vendora.json"=dword:00000000
+"C:\windows\system32\vendorb_vk.json"=dword:00000001
+"C:\windows\system32\vendorc_icd.json"=dword:00000000
+```
+
+In this case, the loader will step through each entry, and check the value.
+If the value is 0, then the loader will attempt to load the file.
+In this case, the loader will open the first and last listings, but not the
+middle.
+This is because the value of 1 for vendorb_vk.json disables the driver.
+
+The Vulkan loader will open each enabled manifest file found to obtain the name
+or pathname of an implementation's shared library (".DLL") file.
+
+Implementations should use the registry locations from the PnP Configuration
+Manager wherever practical.
+Typically, this is most important for ICDs, and the location clearly ties the
+ICD to a given device.
+The `SOFTWARE\Khronos\Vulkan\Drivers` location is the older method for locating
+ICDs, but is primary for non-ICD implementations.
+
+See the
+[Implementation Manifest File Format](#implementation-manifest-file-format)
+section for more details.
+
+
+### Implementation Discovery on Linux
+
+On Linux, the Vulkan loader will scan for implementation manifest files using
+environment variables or corresponding fallback values if the corresponding
+environment variable is not defined:
+
+<table style="width:100%">
+  <tr>
+    <th>Search Order</th>
+    <th>Directory/Environment Variable</th>
+    <th>Fallback</th>
+    <th>Additional Notes</th>
+  </tr>
+  <tr>
+    <td>1</td>
+    <td>$XDG_CONFIG_HOME</td>
+    <td>$HOME/config</td>
+    <td><b>This path is ignored when running an application with root or
+        super-user access</b>.<br/>
```

Comment:
Thanks for the clarification.  We're using secure_getenv based on requests from Chromium, so it sounds like my wording is wrong and should be what you suggest in the first sentence.

Reply:
Am I right in thinking that every time Vulkan-Loader looks at an environment variable, including `$XDG_*`, `$VULKAN_*` and `$HOME`, it's using `secure_getenv()` or equivalent?

If yes, then it might be better to say that *once*, as a general statement about environment variables, rather than putting it in a subset of the places where environment variables are mentioned (and perhaps making people think that other environment variable references are done unconditionally)?

Reply:
I do have a global mention of it as well as individual locations.  The intent was people tend to focus solely on the sections they're looking at and by mentioning it multiple times, they may see the comment in one of those locations.  If you look in the top-level LoaderInterfaceArchitecture.md file, you'll see that there's now a "Elevated Privileges Caveats" section.  I can try to specifically call out the XDG and other items there if you believe it's useful.